Cockpit 275 (#558)
cockpit-project/cockpit-project.github.io: _posts/2022-08-24-cockpit-275.md

---
title: Cockpit 275
author:
date: '2022-08-24'
tags: cockpit, podman
slug: cockpit-275
category: release
summary: ''
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 275 and cockpit-podman 53:

## Metrics: Add read/write usage per disk

State: **OPEN** (will not be included) @ <https://github.com/cockpit-project/cockpit/pull/17657>
{:.warn}

Adds popover with read/write stats for all disks. Fixes bug where paritions on
NVMe are not skipped which results in incorrect read/write stats.

![screenshot of add read/write usage per disk]({{ site.baseurl }}/images/275-metrics-add-read-write-usage-per-disk.png)

The button is hidden when only 1 disk is found.
![screenshot of add read/write usage per disk]({{ site.baseurl }}/images/275-metrics-add-read-write-usage-per-disk-2.png)

On phone:
![screenshot of add read/write usage per disk]({{ site.baseurl }}/images/275-metrics-add-read-write-usage-per-disk-3.png)

## Shell: Support for alternatives to sudo

State: **merged** @ <https://github.com/cockpit-project/cockpit/pull/17536>
{:.note}

> Cockpit always had the ability to run arbitrary programs to gain administrative access, by adding them to a package manifest.  However, the UI would always use "sudo". Now the UI lets you select which program to use when multiple are defined in the manifests.

Reviewer: ```suggestion
Cockpit has always had the theoretical ability to run arbitrary programs to gain administrative access, by adding them to a package manifest.  However, the UI would always use "sudo". Now, if multiple programs are defined, the UI will prompt for which one to use.
```

> By default, Cockpit will still only use "sudo", so unless you actually add more programs via a manifest, there will be no visible change in the UI.

Reviewer: Do we want to link to the youtube video for this?

Author: Do we usually link to videos?

Reviewer 2: The video is not that great... We could point people to the new documentation for the "label" property of a bridge.

> ## Image history

Reviewer: ## cockpit-podman: Image history

Author: We use `Images` & `Podman` in older release notes so I kept it consistent. Maybe _scripts/generate-release-notes should prefix.

State: **merged** @ <https://github.com/cockpit-project/cockpit-podman/pull/1074>
{:.note}

> The image detail view now has a new tab which can show the image history similar to `podman image history`.

Reviewer: ```suggestion
The image detail view now has a new tab which shows the image history similar to `podman image history`.
```

![screenshot of ## image history]({{ site.baseurl }}/images/275-image-history.png)

> ## Restart policy available for users with lingering enabled

Reviewer: ditto, as above

State: **merged** @ <https://github.com/cockpit-project/cockpit-podman/pull/1039>
{:.note}

Although lingering is not recommended, cockpit-podman now supports setting a restart policy when a user has enabled lingering.

## Display ports, volumes and environment variables

State: **merged** @ <https://github.com/cockpit-project/cockpit-podman/pull/1017>
{:.note}

Published ports, mounted volumes and environment variables are now visible in container integration tab.

> By default only variables set by user are shown and after clicking `Show more` all variables are visible.

Reviewer: ```suggestion
By default only variables set by user are shown.  After clicking `Show more` all variables are visible.
```
